### PR TITLE
changes in app.js

### PR DESCRIPTION
**Pull request by kedharvignesh**

### CODE REVIEW

File: chat box - react/src/App.js [hunk 1]
```diff
@@ -14,24 +14,23 @@ const generateSampleMessages = async () => {
...
+    messages.push(message);
   }
   return messages;
 }
 
 function App() {
   const [messages, setMessages] = useState([]);
-  const [enter, setEnter] = useState(true);
+  let canEnter = useRef();
+  canEnter.current = true;
```
Comment (mohantejach) [inline chat box - react/src/App.js:25]: why are you setting it to true here ?

Comment (mohantejach) [reply chat box - react/src/App.js:25]: then it is same as adding a variable without useRef

check the doc and see how to set the initial value

Comment (kedharvignesh) [reply chat box - react/src/App.js:25]: first time while entering the hande scroll function it has to be true la


> why are you setting it to true here ?

first time while entering the hande scroll function it has to be true la

File: chat box - react/src/App.js [hunk 2]
```diff
@@ -14,24 +14,23 @@ const generateSampleMessages = async () => {
...
 
   useEffect(() => {
-    generateSampleMessages().then((randomMessages) => {
-      randomMessages.forEach(msg => {
-        messages.unshift(msg);
-      })
-      setMessages([...messages])
+    generateSampleMessages().then((newMessages) => {
+      newMessages.concat(messages)
+      setMessages([...newMessages]);
```
Comment (mohantejach) [inline chat box - react/src/App.js:33]: I told you to use setMessages((oldMessages) => [...newMessages, ...oldMessages]) right

File: chat box - react/src/App.js [hunk 3]
```diff
@@ -40,33 +39,31 @@ function App() {
 
 
   const storeMessage = (message) => {
-    messages.push(message);
-    setMessages([...messages]);
+    setMessages([...messages, message]);
```
Comment (mohantejach) [inline chat box - react/src/App.js:42]: here also , use arrow function inside setstate

File: chat box - react/src/App.js [hunk 4]
```diff
@@ -40,33 +39,31 @@ function App() {
...
-        if (res) {
-          setEnter(true);
-          res.forEach(msg => {
-            messages.unshift(msg);
-          })
-          setMessages([...messages]);
+        generateSampleMessages().then((newMessages) => {
+          canEnter.current = true;
+          newMessages.concat(messages)
+          setMessages([...newMessages, ...messages])
```
Comment (mohantejach) [inline chat box - react/src/App.js:57]: do as I did above

Comment (kedharvignesh) [reply chat box - react/src/App.js:57]: will make the changes

Comment (mohantejach) [reply chat box - react/src/App.js:57]: don't use message value directly, in some cases you can, but it is not a good practice to write it like that

**Pull request merged**